localstack-core/localstack/services/cloudformation/provider.py

```diff
@@ -269,12 +268,15 @@ def create_stack(self, context: RequestContext, request: CreateStackInput) -> Cr
             state.stacks[stack.stack_id] = stack
             return CreateStackOutput(StackId=stack.stack_id)
 
+        # HACK: recreate the stack (including all of its confusing processes in the __init__ method
```

Comment:
This comment is to add clarity more than anything else. The movement of the construction of the stack is not relevant here